### PR TITLE
Initial collection init

### PR DESCRIPTION
**Pull request by Akasurde**
* Add docs
* Add changelogs
* Update docs

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

**Comment by Akasurde** [comment]
@mukultaneja @goneri @sky-joker @mariolenz @Tomorrow9 @warthog9 Could you please review this? Thanks.

**Comment by Akasurde** [comment]
@mukultaneja Are you interested in this PR? Let me know otherwise I will close it.

**Comment by sky-joker** [comment]
**FYI**

In the Ansible collections, the module documents and README have been automatically generating with [collection_prep](https://github.com/ansible-network/collection_prep).  
I think the efficiency can be improved for generating documents by using collection_prep when changing this repo to the collection.

@Akasurde 
If my comment is wrong, can you please point it out?

**Comment by ansiblejunky** [comment]
This PR would be really useful for many customers we have at Red Hat. Bumping this PR as it should be merged in as soon as possible.

**Comment by Akasurde** [comment]
@sky-joker The documentation is created using `collection_prep` repo only. Thanks for bringing that up.

**Comment by mukultaneja** [comment]
@Akasurde @ansiblejunky @all, Sorry for the delayed response. Thanks for raising a PR but I am having below two concerns. Please help me get the clarity,

1. It contains 49 files that are very hard to review at a time. Can we divide them across PRs?
2. Why do we need to convert these modules into collections. What usecase are we targetting here?

**Comment by Akasurde** [comment]
> @Akasurde @ansiblejunky @ALL, Sorry for the delayed response. Thanks for raising a PR but I am having below two concerns. Please help me get the clarity,
> 
>     1. It contains 49 files that are very hard to review at a time. Can we divide them across PRs?
> 

OK. Here, I am just changing the directory structure of repo. The module code and everything else remain the same (with minor changes).

>     2. Why do we need to convert these modules into collections. What usecase are we targetting here?

The collection is a new way of delivering Ansible content like modules, plugins, roles and playbooks. Once collection is created it can be uploaded to `https://galaxy.ansible.com` from where users, vendors, customers can download and use collection. Please refer https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#collections and https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html

**Comment by ansiblejunky** [comment]
@mukultaneja @Akasurde 
Just to add to what @Akasurde stated above, customers currently want to source this repo to use the modules however they are forced to either refactor the repo into an Ansible Role format to be consumed and used by their playbooks, or convert the repo to what is now considered the "path forward for Ansible" and create a collection.  The collection is a namespace-driven concept for Ansible that handles playbooks, roles, modules, docs, etc within a single repo. Moving to a collection structure additionally requires some minor changes to the module python code because these modules will need to be referenced with namespace-prefix instead of simply the module name itself.  So @Akasurde needed to make minor changes to conform the modules to use the namespace `vmware`.  

There is yet another reason to get this repo moved to a collection and inside our Ansible Galaxy site. Customers currently are not able to source a collection repo from a standard git repo using Ansible 2.9.x. They are only able to pull a collection from Ansible Galaxy (by referencing the namespace). So a recent customer hit this problem and so I see a big benefit to moving this repo to a standard Collection format and into Ansible Galaxy to broaden the customer consumption. 

So perhaps you can run this branch through basic module tests you might have, to ensure the modules are still functioning properly, but other than that I believe @Akasurde mostly generated documentation and moved files around to conform to the collection structure.

**Comment by mukultaneja** [comment]
> @mukultaneja @Akasurde
> Just to add to what @Akasurde stated above, customers currently want to source this repo to use the modules however they are forced to either refactor the repo into an Ansible Role format to be consumed and used by their playbooks, or convert the repo to what is now considered the "path forward for Ansible" and create a collection. The collection is a namespace-driven concept for Ansible that handles playbooks, roles, modules, docs, etc within a single repo. Moving to a collection structure additionally requires some minor changes to the module python code because these modules will need to be referenced with namespace-prefix instead of simply the module name itself. So @Akasurde needed to make minor changes to conform the modules to use the namespace `vmware`.
> 
> There is yet another reason to get this repo moved to a collection and inside our Ansible Galaxy site. Customers currently are not able to source a collection repo from a standard git repo using Ansible 2.9.x. They are only able to pull a collection from Ansible Galaxy (by referencing the namespace). So a recent customer hit this problem and so I see a big benefit to moving this repo to a standard Collection format and into Ansible Galaxy to broaden the customer consumption.
> 
> So perhaps you can run this branch through basic module tests you might have, to ensure the modules are still functioning properly, but other than that I believe @Akasurde mostly generated documentation and moved files around to conform to the collection structure.

sure @ansiblejunky, thanks for the explanation. I can get your point and give me a little time to understand the changes and validate them locally.  It is new and entirely different from the existing repo structure. I am only concern about It would change the structure and usage overnight, hope you can get my point here. Cant, we look for a way where we could maintain both the structure like through namespace or something?

**Comment by mariolenz** [comment]
I agree with @ansiblejunky: In the long run, you should aim to make this a collection. After all, collections are the default way to deliver ansible content (modules, roles and so on). But how (and when) you get there, that's up to you.

> It is new and entirely different from the existing repo structure.

I have a little problem with the word "new" here, [collections](https://docs.ansible.com/ansible/2.9/dev_guide/developing_collections.html) have been introduced with [ansible 2.9](https://pypi.org/project/ansible/2.9.0/) one and a half year ago (october 2019, I think). And they've been announced even earlier...

> I am only concerned about the structure and change in usage overnight, hope you can get my point here.

Yes, of course. Breaking your users playbooks wouldn't be a good idea at all.

> Cant, we look for a way where we could maintain both the structure like through namespace or something?

Since @Akasurde states that "I am just changing the directory structure of repo. The module code and everything else remain the same (with minor changes)", maybe it's possible to keep the old structure by just linking? So that, for example, `modules/vcd_catalog_item.py` would be a link to `plugins/modules/vcd_catalog_item.py`. I'm not sure if the "minor changes" are really minor enough to allow this, but at least it's an idea.

**Comment by ansiblejunky** [comment]
I personally wouldn't do links in the repo. The easiest way to handle "deprecation" of the current folder/file structure is to tag/version it and allow current users to simply clone that specific tag/version. This seems the simplest method and uses good versioning methods. It also allows the repo to mature, which it should. As @mariolenz stated, collections is not a new concept.

This model also allows you to backport any new features/bug-fixes to the tagged code whenever you feel it's appropriate to do so. And you eventually state a code freeze date when you no longer backport.

**Comment by Akasurde** [comment]
@mukultaneja Like @ansiblejunky suggested you can create a stable branch to maintain the old directory structure. And create a `devel` branch to keep the new world directory structure.

**Comment by mariolenz** [comment]
I agree, having two branches (one for the new collection, one for the legacy non-collection code) would be the cleanest solution. And then deprecate the non-collection branch and stop it some time in the future.

File: README.md [hunk 1]
```diff
@@ -1,69 +1,104 @@
...
-11. vcd_vapp_vm
-12. vcd_vapp_vm_disk
-13. vcd_vapp_vm_nic
-14. vcd_vapp_vm_snapshot
-15. vcd_vdc_gateway
-16. vcd_vdc_network
-17. vcd_gateway_services
+<!--end collection content-->
+
+## Testing and Development
```
Comment (mukultaneja) [inline README.md:75]: We don't have this testing setup for vcd yet

File: README.md [hunk 2]
```diff
@@ -1,69 +1,104 @@
...
+- Refresh the README.md: `tox -e add_docs`
+- Refresh the changelog: `tox -e antsibull-changelog -- release --verbose --version 1.2.0`
+- Clean up the changelog fragments.
+- Commit everything and push a PR for review
+
+Push the release:
+- Tag the release: `git tag -s 1.0.0`
+- Push the tag: `git push origin 1.0.`
+
+## Communication
```
Comment (mukultaneja) [inline README.md:97]: We don't have any channel for VCD yet

**Comment by mukultaneja** [comment]
@here, I looked at the changes and they look good to me except for the above comments and `docs` section. We don't want to manage the `docs` folder instead would like to continue using GitHub [wiki ](https://github.com/vmware/ansible-module-vcloud-director/wiki)to provide documentation on the modules. So it would be better if we remove it from the PR.

To manage both types of repo structure, I think it would be a good idea to go with a different branch for now. I would create a new branch named `collection` and I can merge this PR with that branch, test it thoroughly and after a while merge it with master if no one raised issues. Hope it sounds good to all.

File: README.md (hunk 2 above)
Comment (Akasurde) [reply README.md:97]: My bad.

File: README.md (hunk 1 above)
Comment (Akasurde) [reply README.md:75]: OK, I will add TBD section instead.

**Comment by Akasurde** [comment]
@mukultaneja Thanks for reviewing. Managing wiki is cumbersome, they easily go out of date. Let us decide on docs in a separate PR since it is not a hard requirement for collection.

**Comment by mukultaneja** [comment]
@Akasurde thanks for updating the PR. I have created a new branch `collection`, could you please open a PR for that branch?

**Comment by Akasurde** [comment]
Done.

**Comment by mukultaneja** [comment]
> Done.

thanks, @Akasurde for putting effort into this PR.

**Pull request merged**